File: 64/js/64.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
var minPathSum = function(grid) {

    // handle base cases
    for (let i = 1; i < grid.length; i++){
       grid[i][0] = grid[i][0] + grid[i - 1][0];
    }

    for (let j = 1; j < grid[0].length; j++){
       grid[0][j] = grid[0][j] + grid[0][j - 1];
    }

    // recursive case
    for (let i = 1; i < grid.length; i++){
        for (let j = 1; j < grid[i].length; j++){
            grid[i][j] = grid[i][j] + Math.min(grid[i - 1][j], grid[i][j - 1]);
        }
    }

    return grid[grid.length - 1][grid[0].length - 1];
};

File: 64/go/64.go
func minPathSum(grid [][]int) int {

    // base cases
    for i := 1; i < len(grid); i++ {
        grid[i][0] = grid[i][0] + grid[i-1][0]
    }

    for j := 1; j < len(grid[0]); j++ {
        grid[0][j] = grid[0][j] + grid[0][j-1]
    }

    // recursive case
    for i := 1; i < len(grid); i++ {
        for j := 1; j < len(grid[i]); j++{
            grid[i][j] = grid[i][j] + minNumber(grid[i-1][j], grid[i][j-1])
        }
    }

    return grid[len(grid) - 1][len(grid[0]) - 1]
}

func minNumber( a int, b int) int {
    if a > b {
        return b
    }
    return a
}